contracts/ResilientOracle.sol

```diff
@@ -329,8 +325,40 @@ contract ResilientOracle is PausableUpgradeable, AccessControlledV8, ResilientOr
         enabled = tokenConfigs[asset].enableFlagsForOracles[uint256(role)];
     }
 
+    /**
+     * @notice Updates the pivot oracle price. Currently using TWAP
+     * @dev Cache the asset price and return if already cached
+     * @param asset asset address
+     */
+    function _updateAssetPrice(address asset) internal {
+        if (Transient.readCachedPrice(CACHE_SLOT, asset) != 0) {
+            return;
+        }
+
+        (address mainOracle, bool mainOracleEnabled) = getOracle(asset, OracleRole.MAIN);
+        if (mainOracle != address(0) && mainOracleEnabled) {
+            //if main oracle is not CorrelatedTokenOracle it will revert so we need to catch the revert
+            try ICappedOracle(mainOracle).updateSnapshot() {} catch {}
+        }
```

Comment:
Would it also make sense to try to update the snapshots of the fallback and pivot oracles?

Reply:
Yes, we can do so, but that will increase the gas usage as well and in the counterpart we are using mainOracle in most of the places so if we skip that would be okay too.